pytest_terraform/tf.py

```diff
@@ -72,8 +72,15 @@ def apply(self, plan=True):
             apply_args = self._get_cmd_args("apply", plan=plan_path)
         elif plan:
             apply_args = self._get_cmd_args("apply", plan="")
-        self._run_cmd(apply_args)
-        return TerraformState.from_file(self.state_path, self)
+        try:
+            self._run_cmd(apply_args)
+            return TerraformState.from_file(self.state_path, self)
+        except subprocess.CalledProcessError as e:
+            try:
+                # Try to destroy partially applied resources
+                self.destroy()
+            finally:
+                raise e from None
 
```

Comment:
Intentionally ignore any errors from Destroy. In the test case, Destroy happens to fail for the same reason as the apply fails. In production scenarios, Destroy will normally work and destroy anything partially created by the apply step.